packages/avatars/src/Avatar.js

```diff
@@ -5,50 +5,65 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+import React from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import { retrieveTheme } from '@zendeskgarden/react-theming';
-import AvatarStyles from '@zendeskgarden/css-avatars';
+import { StyledAvatar, StyledBadge } from './styled';
 
-const COMPONENT_ID = 'avatars.avatar';
 const SIZE = {
   EXTRASMALL: 'extrasmall',
   SMALL: 'small',
   LARGE: 'large'
 };
 
+const STATUS = {
+  AVAILABLE: 'available',
+  ACTIVE: 'active',
+  AWAY: 'away'
+};
+
 /**
- * Accepts all `<figure>` props
+ * Accepts all `<figure>` attributes and events
  */
-const Avatar = styled.figure.attrs({
-  'data-garden-id': COMPONENT_ID,
-  'data-garden-version': PACKAGE_VERSION,
-  className: props =>
-    classNames(AvatarStyles['c-avatar'], {
-      // Types
-      [AvatarStyles['c-avatar--system']]: props.system,
-
-      // Sizes
-      [AvatarStyles['c-avatar--xs']]: props.size === SIZE.EXTRASMALL,
-      [AvatarStyles['c-avatar--sm']]: props.size === SIZE.SMALL,
-      [AvatarStyles['c-avatar--lg']]: props.size === SIZE.LARGE,
-
-      // State
-      [AvatarStyles['is-disabled']]: props.disabled,
-      [AvatarStyles['c-avatar--borderless']]: props.isBorderless
-    })
-})`
-  ${props => retrieveTheme(COMPONENT_ID, props)};
-`;
+const Avatar = ({ isSystem, size, status, children, badge, ...other }) => {
+  let computedStatus = status;
+
+  if (status === STATUS.AVAILABLE && badge !== undefined) {
+    computedStatus = STATUS.ACTIVE;
+  }
+
+  const FormattedBadge = () => {
+    if (status === undefined) {
+      return null;
+    }
+
+    if (computedStatus === STATUS.AWAY) {
+      return <StyledBadge />;
+    }
+
+    return <StyledBadge>{badge}</StyledBadge>;
+  };
+
+  return (
+    <StyledAvatar
+      isSystem={isSystem}
+      size={size}
+      status={computedStatus}
+      aria-live="polite"
+      {...other}
+    >
+      {children}
+      <FormattedBadge />
+    </StyledAvatar>
+  );
+};
 
 Avatar.propTypes = {
   /** Applies system styling */
-  system: PropTypes.bool,
+  isSystem: PropTypes.bool,
```

Comment:
is it really important to change this property name here? If so, then it will need to be noted in the BREAKING CHANGE commit merge message.

Reply:
For boolean props we need to watch out for valid attribute names. We haven't done this in the past. If we follow the `isFoo` pattern consistently we can avoid this.

We should probably do this system wide as we move to CSS-in-JS, but I thought this would be a good place to start.